src/app/utils/index.ts

```diff
@@ -151,3 +166,121 @@ export function b64EncodeUnicode(str: string) {
 export function stringToDataURL(mimeType: string, content: string) {
   return "data:" + mimeType + ";base64," + b64EncodeUnicode(content);
 }
+
+
+function checkConvertion(type: DataType, dataSample: Array<string | boolean | Date | number>) {
+  let convertable = true;
+  if (type === DataType.String) {
+    return convertable;
+  }
+
+  switch (type) {
+    case DataType.Boolean:
+      for (const data of dataSample) {
+        if (data.toString() != "0" && data.toString() != "true" && data.toString() != "1" && data.toString() != "false") {
```

Comment:
Can `data` be null/undefined here?  I know you marked your `dataSample` like it couldn't be, but sometimes types can be wrong :)

Reply:
Same question for each of the cases below

Reply:
Add checks